Process: Support configuring log level keywords use

LowEQ opened the pull request:
initial implementation, still need add `atest`  probably to `atest\testdata` and `atest\robot`

pekkaklarck commented on src/robot/libraries/Process.py:
```diff
@@ -323,7 +323,7 @@ def __init__(self):
         self._results = {}
 
-    def run_process(self, command, *arguments, **configuration):
+    def run_process(self, command, log_level=None, *arguments, **configuration):
```
`log_level` cannot be a positional argument because it would break pretty much all current usages of this keyword. It should be a keyword-only argument, but I'm not sure is it better to be a separate argument or should it be passed in in `**configuration` like other such options. The reason we've used `**configuration` instead of real keyword-only arguments is that this code was written originally for Python 2 that didn't even have keyword-only arguments. It probably should be refactored now, but that should be done separately from this PR. In the current situation including it in `**configuration` is probably easiest and most consistent.

LowEQ replied:
I am not sure, if it is properly done  by just adding `log_level` through `ProcessConfiguration` `__init__`.

pekkaklarck commented on src/robot/libraries/Process.py:
```diff
@@ -349,26 +349,38 @@ def run_process(self, command, *arguments, **configuration):
+        | Should Be Equal | ${result.stdout} | Hello, world! |
+        | ${result} = | Run Process | ${command} | 'debug' | stdout=${CURDIR}/stdout.txt | stderr=STDOUT |
+        | ${result} = | Run Process | ${command} | 'Warn' | timeout=1min | on_timeout=continue |
+        | ${result} = | Run Process | java -Dname\\=value Example | 'Debug' | shell=True | cwd=${EXAMPLE} |
```
There's no need to duplicate all examples with `log_level`. Adding a new example with it, or possibly just adding it to one of the existing examples, is enough.

LowEQ commented:
tried run acceptance tests, all failed, rolling back this branch changes, and try to add the log_level arg into `**configuration`

LowEQ committed:
change implementation and fixed most of test fails

LowEQ committed:
fixed failed atests

LowEQ committed:
adding atests

LowEQ commented:
added atests

LowEQ committed:
Merge branch 'robotframework:master' into Process--Support-configuring-log-level-keywords-use

LowEQ committed:
Merge branch 'robotframework:master' into Process--Support-configuring-log-level-keywords-use

franzhaas commented:
this interacts with this one https://github.com/robotframework/robotframework/pull/5302. 

It is likely that there will be some coordination necessary.

LowEQ committed:
Merge branch 'master' into Process--Support-configuring-log-level-keywords-use

LowEQ commented:
> this interacts with this one #5302.
> 
> It is likely that there will be some coordination necessary.

sorry, I was in an long vacation, will try to merge these changes together

LowEQ committed:
fixing tests for merge

LowEQ committed:
tests fix

LowEQ commented:
Should be able to review again. when running locally one of the Unit test failed for some reason, but after pushing the change the unit tests passes all?

LowEQ committed:
changed documentation for Framework 7.4.

LowEQ committed:
change version number

LowEQ commented:
> Now that RF 7.4 is getting ready, I finally had a possibility to look at this PR. Unfortunately there are lot of conflicts, most likely due to #5412, that need to be resolved before this could be merged. I did, however, take a quick look at the changes. They look good in general, but there are various bigger and smaller things that would need to be fixed.
> 
> Are you @LowEQ still around and interested on this PR? If you are, I can do a proper review for the current changes later this week. In addition to addressing the review comments, you then needed to resolve the conflicts.

yeah, just got time read email this week in detail, Sure, I got time to look at this PR and do needed resolves and changes, as soon as I have done it. I can again request an review and following fix based on the review.

LowEQ committed:
Merge branch 'master' into Process--Support-configuring-log-level-keywords-use

LowEQ committed:
fixing merge errors

LowEQ committed:
resolve error fix 2

LowEQ committed:
fixing failed test + adding needed test case change and documentation

pekkaklarck commented:
Issue #4929 was closed as not planned so this PR needs to closed as well. There are also conflicts due to somewhat recently added type hints that needed to be resolved, but that doesn't really matter at this point.
